impl/sdl/sdl.odin

```diff
@@ -43,103 +43,112 @@ setup_state :: proc(using state: ^SDL_State) {
-            if e.button.button == u8(sdl.Mousecode.Right)  do state.mouse_down[1] = true;
-            if e.button.button == u8(sdl.Mousecode.Middle) do state.mouse_down[2] = true;
+        case .MOUSEBUTTONDOWN: {
+            if e.button.button == u8(sdl.BUTTON_LEFT)   do state.mouse_down[0] = true;
+            if e.button.button == u8(sdl.BUTTON_RIGHT)  do state.mouse_down[1] = true;
+            if e.button.button == u8(sdl.BUTTON_MIDDLE) do state.mouse_down[2] = true;
         }
 
-        case .Key_Down, .Key_Up: {
+        case .KEYDOWN, .KEYUP: {
+
             sc := e.key.keysym.scancode;
-            io.keys_down[sc] = e.type == .Key_Down;
-            io.key_shift = sdl.get_mod_state() & (sdl.Keymod.LShift|sdl.Keymod.RShift) != nil;
-            io.key_ctrl  = sdl.get_mod_state() & (sdl.Keymod.LCtrl|sdl.Keymod.RCtrl)   != nil;
-            io.key_alt   = sdl.get_mod_state() & (sdl.Keymod.LAlt|sdl.Keymod.RAlt)     != nil;
+            io.keys_down[sc] = e.type == .KEYDOWN;
+			mod_state := sdl.GetModState()
+			io.key_ctrl = (sdl.KeymodFlag.LCTRL in mod_state) | (sdl.KeymodFlag.RCTRL in mod_state)
+			io.key_shift = (sdl.KeymodFlag.LSHIFT in mod_state) | (sdl.KeymodFlag.RSHIFT in mod_state)
+			io.key_alt = (sdl.KeymodFlag.LALT in mod_state) | (sdl.KeymodFlag.RALT in mod_state)
 
-            when ODIN_OS == "windows" {
+            when ODIN_OS == .Windows{
                 io.key_super = false;
             } else {
-                io.key_super = sdl.get_mod_state() & (sdl.Keymod.LGui|sdl.Keymod.RGui) != nil;
+                io.key_super = (sdl.KeymodFlag.LGUI in mod_state) | (sdl.KeymodFlag.RGUI in mod_state)//i32(transmute(u16)(sdl.GetModState())) & i32(sdl.Keycode.LGUI | sdl.Keycode.RGUI) != 0;
             }
+			
+			fmt.println(io.key_super)
         }
     }
 }
 
 update_dt :: proc(state: ^SDL_State) {
     io := imgui.get_io();
-    freq := sdl.get_performance_frequency();
-    curr_time := sdl.get_performance_counter();
+    freq := sdl.GetPerformanceFrequency();
+    curr_time := sdl.GetPerformanceCounter();
     io.delta_time = state.time > 0 ? f32(f64(curr_time - state.time) / f64(freq)) : f32(1/60);
     state.time = curr_time;
 }
 
 update_mouse :: proc(state: ^SDL_State, window: ^sdl.Window) {
     io := imgui.get_io();
     mx, my: i32;
-    buttons := sdl.get_mouse_state(&mx, &my);
-    io.mouse_down[0] = state.mouse_down[0] || (buttons & u32(sdl.Mousecode.Left))   != 0;
-    io.mouse_down[1] = state.mouse_down[1] || (buttons & u32(sdl.Mousecode.Right))  != 0;
-    io.mouse_down[2] = state.mouse_down[2] || (buttons & u32(sdl.Mousecode.Middle)) != 0;
+    buttons := sdl.GetMouseState(&mx, &my);
+    io.mouse_down[0] = state.mouse_down[0] || (buttons & u32(sdl.BUTTON_LEFT))   != 0;
+    io.mouse_down[1] = state.mouse_down[1] || (buttons & u32(sdl.BUTTON_RIGHT))  != 0;
+    io.mouse_down[2] = state.mouse_down[2] || (buttons & u32(sdl.BUTTON_MIDDLE)) != 0;
```

Comment:
In testing I found this to be incorrect, this needs to be changed to `BUTTON_LMASK`, `BUTTON_RMASK` and `BUTTON_MMASK`.